todo_app/app.py

```diff
@@ -1,75 +1,78 @@
-    due_date = request.form.get('task-due-date')
-    if due_date != "":
-        due_date = datetime.strptime(request.form.get('task-due-date'), "%d/%m/%Y")
-    create_task(request.form.get('to-do-title'), due_date, request.form.get('to-do-notes'))
-    return redirect('/')
+    @app.route('/', methods=['POST'])
+    def add_to_do():
+        due_date = request.form.get('task-due-date')
+        if due_date != "":
+            due_date = datetime.strptime(request.form.get('task-due-date'), "%d/%m/%Y")
+        create_task(request.form.get('to-do-title'), due_date, request.form.get('to-do-notes'))
+        return redirect('/')
+
+    @app.route('/<status>', methods=['POST'])
+    def add_to_do_from_filtered_tasks(status):
+        add_to_do()
+        return redirect('/')
 
-@app.route('/<status>', methods=['POST'])
-def add_to_do_from_filtered_tasks(status):
-    add_to_do()
-    return redirect('/')
+    @app.route('/task/<id>')
+    def view_to_do_item(id):
+        task = get_task(id)
+        return render_template("task.html", task=task)
 
-@app.route('/task/<id>')
-def view_to_do_item(id):
-    task = get_task(id)
-    return render_template("task.html", task=task)
+    @app.route('/task/<id>', methods=['POST'])
+    def amend_item(id):
+        amended_status = request.form.get('task-status')
+        amended_title = request.form.get('task-title')
+        amended_notes = request.form.get('task-notes')
+        date = request.form.get('task-due-date')
+        amended_due_complete = "true" if amended_status == "Complete" else "false"
+        amended_due_date = datetime.strptime(date, "%d/%m/%Y") if date != "" else ""
+        
+        task = Item(id, amended_title, amended_status, amended_due_complete, amended_due_date, None, amended_notes)
+        edit_task(task)
+        return redirect('/task/' + id)
 
-@app.route('/task/<id>', methods=['POST'])
-def amend_item(id):
-    amended_status = request.form.get('task-status')
-    amended_title = request.form.get('task-title')
-    amended_notes = request.form.get('task-notes')
-    date = request.form.get('task-due-date')
-    amended_due_complete = "true" if amended_status == "Complete" else "false"
-    amended_due_date = datetime.strptime(date, "%d/%m/%Y") if date != "" else ""
-    
-    task = Item(id, amended_title, amended_status, amended_due_complete, amended_due_date, amended_notes)
-    edit_task(task)
-    return redirect('/task/' + id)
+    @app.route('/delete/<id>')
+    def view_delete_item(id):
+        task = get_task(id)
+        if task != None:
+            return render_template("deletetask.html", task=task)
+        else:
+            return redirect('/')
 
-@app.route('/delete/<id>')
-def view_delete_item(id):
-    task = get_task(id)
-    if task != None:
-        return render_template("deletetask.html", task=task)
-    else:
+    @app.route('/delete/<id>', methods=['POST'])
+    def delete_task(id):
+        if request.form.get('delete-task-button'):
```

Comment:
Oddly enough, adding that line was the only way to prevent the app from calling the delete request three times. Once I remove it, deleting a task would try to send the request three times and cause an error. 🤷